test/graphql-api/link.test.ts

```diff
@@ -48,13 +55,9 @@ test('mocks a GraphQL query to the GitHub GraphQL API', async () => {
       },
     },
   })
-
-  await runtime.cleanup()
```

Comment:
I would not advise moving this teardown to the `beforeAll`/`afterAll` hooks. This may be a personal preference, but I prefer `runtime` to be explicitly declared and cleaned within each test. 

Take a look at [Avoid nesting when you're testing](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) by Kent C. Dodds. I think it has good points as to why I've adopted an explicit `runtime` declaration approach.

Reply:
Removing this change will also help us keep the pull request changes scoped, which is always a good thing to do. If we discover that using jest hooks would improve our tests, we can always roll this out as a separate dedicated pull request.

Reply:
OK, I'll remove that change. Just my curiosity, what do you think about resource leaks when a test fails?

Reply:
That's a viable concern, as an exception during the test instructions (i.e. originating from a failed assertion) will short circuit the test function and the cleanup phase won't ever get called.

That being said, I don't think that before/after hooks are the solution to that. Not unless you want to ditch the "shallow" tests principle. 

That's the main reason I've written [`flat-test`](https://github.com/kettanaito/test-flat). It allows to have shallow flat tests while each test accepts a before and after steps that always execute and can share context between each other (one of the pain points of having to mutate things in a conventional before/after setup). We may consider migrating MSW test suites to use `flat-test`.